test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java

```diff
@@ -120,40 +127,74 @@ public String getID() {
+                        .put("ledgerVolumeClaimTemplate", bkPersistentVolumeSpec)
+                        .put("journalVolumeClaimTemplate", bkPersistentVolumeSpec)
+                        .build())
+                .put("autoRecovery", true)
+                .build();
 
         final Map<String, Object> pravegaSpec = ImmutableMap.<String, Object>builder().put("controllerReplicas", controllerCount)
-                                                                                      .put("segmentStoreReplicas", segmentStoreCount)
-                                                                                      .put("debugLogging", true)
-                                                                                      .put("cacheVolumeClaimTemplate", pravegaPersistentVolumeSpec)
-                                                                                      //.put("options", props.getProperties())
-                                                                                      .put("options", props)
-                                                                                      .put("image",
-                                                                                       getImageSpec(DOCKER_REGISTRY + PREFIX + "/" + PRAVEGA_IMAGE_NAME, PRAVEGA_VERSION))
-                                                                                      .put("tier2", tier2Spec())
-                                                                                      .build();
+                .put("segmentStoreReplicas", segmentStoreCount)
+                .put("debugLogging", true)
+                .put("cacheVolumeClaimTemplate", pravegaPersistentVolumeSpec)
+                .put("controllerResources", getResources("2000m", "3Gi", "1000m", "1Gi"))
+                .put("segmentStoreResources", getResources("2000m", "5Gi", "1000m", "3Gi"))
+                .put("options", props)
+                .put("image", pravegaImgSpec)
+                .put("tier2", tier2Spec())
+                .build();
+
         return ImmutableMap.<String, Object>builder()
                 .put("apiVersion", "pravega.pravega.io/v1alpha1")
                 .put("kind", CUSTOM_RESOURCE_KIND_PRAVEGA)
                 .put("metadata", ImmutableMap.of("name", PRAVEGA_ID, "namespace", NAMESPACE))
-                .put("spec", ImmutableMap.builder().put("zookeeperUri", zkLocation)
-                                         .put("bookkeeper", bookkeeperSpec)
-                                         .put("pravega", pravegaSpec)
-                                         .build())
+                .put("spec", pravegaClusterSpecBuilder(zkLocation, bookkeeperSpec, pravegaSpec))
                 .build();
     }
 
+    protected Map<String, Object> pravegaClusterSpecBuilder(String zkLocation, Map<String, Object> bookkeeperSpec, Map<String, Object> pravegaSpec) {
+        if (isLaterOperator) {
+            return ImmutableMap.<String, Object>builder()
```

Comment:
You could have 1 map with common entries like:
```
ImmutableMap<String, String> commonEntries = ImmutableMap.<String, Object>builder()
                    .put("zookeeperUri", zkLocation)
                    .put("bookkeeper", bookkeeperSpec)
                    .put("pravega", pravegaSpec)
                    .build();
```
and then for 0.4.0 spec, just do: 
```
ImmutableMap<String, String> immutableMap = ImmutableMap.<String, String>builder()
  .putAll(commonEntries)
  .put("version", PRAVEGA_VERSION)
  .build();
```

Reply:
done